odl/solvers/functional/default_functionals.py

```diff
@@ -1326,6 +1376,185 @@ def convex_conj(self):
+        dest = [a + arr.ndim if a < 0 else a for a in dest]
+
+        order = [n for n in range(arr.ndim) if n not in source]
+
+        for dest, src in sorted(zip(dest, source)):
+            order.insert(dest, src)
+
+        return arr.transpose(order)
+
+    def _asarray(self, vec):
+        """Convert ``x`` to an array.
+
+        Here the indices are changed such that the "outer" indices come last,
+        this is in order to have the access order as `numpy.linalg.svd` needs
+        them.
+
+        This is the inverse of `_asvector`.
+        """
+        shape = self.domain[0][0].shape + self.pshape
+        arr = np.zeros(shape, dtype=self.domain.dtype)
+        for i, xi in enumerate(vec):
+            for j, xii in enumerate(xi):
+                arr[..., i, j] = xii.asarray()
+
+        return arr
+
+    def _asvector(self, arr):
+        """Convert ``vec`` to an `domain` element.
+
+        This is the inverse of `_asarray`.
+        """
+
+        result = self._moveaxis(arr, [-2, -1], [0, 1])
+
+        return self.domain.element(result)
+
+    def _call(self, x):
+        """Return ``self(x)``."""
+
+        # Convert to array with most
+        arr = self._asarray(x)
+        svd_diag = np.linalg.svd(arr, compute_uv=False)
+
+        # Rotate the axes so the svd-direction is first
+        s_reordered = self._moveaxis(svd_diag, -1, 0)
+
+        # Return nuclear norm
+        return self.outernorm(self.pwisenorm(s_reordered))
+
+    @property
+    def proximal(self):
+        """Return the proximal operator.
+
+        Raises
+        ------
+        NotImplementedError
+            if ``outer_exp`` is not 1 or ``singular_vector_exp`` is not 1, 2 or
+            infinity
+        """
+
+        if self.outernorm.exponent != 1:
+            raise NotImplementedError('`proximal` only implemented for '
+                                      '`outer_exp==1`')
+        if self.pwisenorm.exponent not in [1, 2, np.inf]:
+            raise NotImplementedError('`proximal` only implemented for '
+                                      '`singular_vector_exp` in [1, 2, inf]')
+
+        def nddot(a, b):
+            """Compute pointwise matrix product in the last indices."""
+            return np.einsum('...ij,...jk->...ik', a, b)
+
+        func = self
+
+        class NuclearNormProximalOperator(Operator):
+            """Proximal operator of `NuclearNorm`."""
+            def __init__(self, sigma):
+                self.sigma = sigma
+                Operator.__init__(self, func.domain, func.domain)
+
+            def _call(self, x):
+                """Return ``self(x)``."""
+                arr = func._asarray(x)
+
+                # Compute SVD
+                U, s, Vt = np.linalg.svd(arr, full_matrices=False)
+
+                # transpose pointwise
+                V = Vt.swapaxes(-1, -2)
+
+                # Take pseudoinverse of s
+                sinv = s.copy()
+                sinv[sinv != 0] = 1 / sinv[sinv != 0]
```

Comment:
Well the pointwise SVD is not really free to compute. Frankly all of this could easily be moved the the GPU at some later point if we want to. It is very GPU friendly.